packages/material-ui/src/ClickAwayListener/ClickAwayListener.js

```diff
@@ -111,14 +122,52 @@ const ClickAwayListener = React.forwardRef(function ClickAwayListener(props, ref
     return undefined;
   }, [handleClickAway, mouseEvent]);
 
-  return <React.Fragment>{React.cloneElement(children, { ref: handleRef })}</React.Fragment>;
+  // Keep track of mouse/touch events that bubbled up through the portal.
+  const handleSyntheticMouse = (event) => {
+    syntheticEventRef.current = true;
+
+    const childrenProps = children.props;
+    if (childrenProps[mouseEvent]) {
+      childrenProps[mouseEvent](event);
+    }
+  };
+
+  const handleSyntheticTouch = (event) => {
+    syntheticEventRef.current = true;
+
+    const childrenProps = children.props;
+    if (childrenProps[touchEvent]) {
+      childrenProps[touchEvent](event);
+    }
+  };
+
+  const childrenProps = {};
+
+  if (mouseEvent !== false) {
+    childrenProps[mouseEvent] = handleSyntheticMouse;
+  }
+
+  if (touchEvent !== false) {
+    childrenProps[touchEvent] = handleSyntheticTouch;
+  }
+
+  return (
+    <React.Fragment>
+      {React.cloneElement(children, { ref: handleRef, ...childrenProps })}
+    </React.Fragment>
+  );
 });
 
 ClickAwayListener.propTypes = {
   /**
    * The wrapped element.
    */
   children: elementAcceptingRef.isRequired,
+  /**
+   * If `true`, the React tree is ignored and only the DOM tree is considered.
+   * This prop changes how portaled elements are handled.
```

Comment:
I think in British English this would be spelt as `portalled`, but for consistency with other occurrences of `portaled` in the repo I used 1 `l`

Reply:
We use American English on the whole repository :).